web-common/src/features/dashboards/url-state/DashboardURLStateSync.svelte

```diff
@@ -117,60 +117,7 @@
 
     // Pressing back button and going back to empty url state should not restore from session store
     const backButtonUsed = type === "popstate";
-    const skipSessionStorage =
-      backButtonUsed && $page.url.searchParams.size === 0;
-
-    let partialExplore = partialExploreStateFromUrl;
-    let shouldUpdateUrl = false;
-    if (exploreStateFromSessionStorage && !skipSessionStorage) {
-      partialExplore = exploreStateFromSessionStorage;
-      shouldUpdateUrl = true;
-    }
-
-    const redirectUrl = new URL(to.url);
-    metricsExplorerStore.mergePartialExplorerEntity(
-      exploreName,
-      partialExplore,
-      metricsSpec,
-    );
-    if (shouldUpdateUrl) {
-      // if we added extra url params from sessionStorage then update the url
-      redirectUrl.search = getUpdatedUrlForExploreState(
-        exploreSpec,
-        timeControlsState,
-        defaultExplorePreset,
-        partialExplore,
-        $page.url.searchParams,
-      );
-    }
-    // update session store when back button was pressed.
-    if (backButtonUsed) {
-      updateExploreSessionStore(
-        exploreName,
-        extraKeyPrefix,
-        $dashboardStore,
-        exploreSpec,
-        timeControlsState,
-      );
-    }
-
-    if (
-      !shouldUpdateUrl ||
-      redirectUrl.search === to.url.toString() ||
-      // redirect loop breaker
-      (prevUrl && prevUrl === redirectUrl.toString())
-    ) {
-      prevUrl = redirectUrl.toString();
-      return;
-    }
-
-    prevUrl = redirectUrl.toString();
-    // using `replaceState` directly messes up the navigation entries,
-    // `from` and `to` have the old url before being replaced in `afterNavigate` calls leading to incorrect handling.
-    void goto(redirectUrl, {
-      replaceState: true,
-      state: $page.state,
-    });
+    void handleAfterNavigate(to.url, backButtonUsed);
```

Comment:
Given all the above code in the `afterNavigate` scope, it's not entirely clear to me what's supposed to be encapsulated in the `handleAfterNavigate` function.

Reply:
Reverted since we dont use async methods